api/http/handler/stacks/create_swarm_stack.go

```diff
@@ -340,7 +340,9 @@ func (handler *Handler) deploySwarmStack(config *swarmStackDeploymentConfig) err
 		return err
 	}
 
-	if !settings.AllowBindMountsForRegularUsers && !isAdminOrEndpointAdmin {
+	if (!settings.AllowDeviceMappingForRegularUsers ||
```

Comment:
No need to add this check for Swarm stacks, device mapping is not supported on Swarm: https://docs.docker.com/compose/compose-file/#devices